### PR TITLE
Visitor can see the cars available for rent on the landing page

### PR DESCRIPTION
**Pull request by Jon1969Edwards**
- Created car card controller
- Created 'Automobile' model
- Updated index action and created the show action
- Created the car card on the landing page
- Added the HAML and Faker gems
- Added a feature test for car card
- Added unit test for car card
- Set routes for automobile
screen shot 2018-12-04 at 4 59 35 pm

PT link:  https://www.pivotaltracker.com/story/show/162393150

### CODE REVIEW

File: spec/models/automobile_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Automobile, type: :model do
+  describe 'DB table' do
+    it { is_expected.to have_db_column :brand }
+    it { is_expected.to have_db_column :model }
+    it { is_expected.to have_db_column :year }
+    it { is_expected.to have_db_column :price }
+  end
```
Comment (GergKllai1) [inline spec/models/automobile_spec.rb:9]: You should test the Factory validity as well

File: spec/factories/automobiles.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :automobile do
+    brand { "MyString" }
+    model { "MyString" }
+    year { "" }
```
Comment (GergKllai1) [inline spec/factories/automobiles.rb]: Year and Price are integers, not strings

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
+  resources :automobile
```
Comment (GergKllai1) [inline config/routes.rb]: Use only those resources which are needed, now only index

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
...
+  root controller: :car_card, action: :index
   devise_for :users
   resources :welcome, only: :index
```
Comment (GergKllai1) [inline config/routes.rb]: Remove the routes and the controller and view file, we don't need this anymore

File: spec/factories/automobiles.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    price { "" }
```
Comment (GergKllai1) [inline spec/factories/automobiles.rb]: Use actual examples, like volvo, 40

**Review by GergKllai1 — CHANGES_REQUESTED** [review]

File: spec/factories/automobiles.rb (hunk 2 above)
Comment (johannaanden) [reply spec/factories/automobiles.rb]: Done

File: spec/models/automobile_spec.rb (hunk 1 above)
Comment (johannaanden) [reply spec/models/automobile_spec.rb:9]: Done

File: config/routes.rb (hunk 3 above)
Comment (johannaanden) [reply config/routes.rb]: Done

File: config/routes.rb (hunk 4 above)
Comment (johannaanden) [reply config/routes.rb]: Done

File: spec/factories/automobiles.rb (hunk 5 above)
Comment (johannaanden) [reply spec/factories/automobiles.rb]: Done

File: app/controllers/car_card_controller.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class CarCardController < ApplicationController
+  def index
+    @automobile = Automobile.all
```
Comment (oliverochman) [inline app/controllers/car_card_controller.rb]: the variable should be in plural

File: app/controllers/car_card_controller.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+class CarCardController < ApplicationController
```
Comment (oliverochman) [inline app/controllers/car_card_controller.rb]: the name of the controller should be the same as the model

**Review by oliverochman — CHANGES_REQUESTED** [review]
Update the title of this PR to "Visitor can see the cars available for rent on the landing page"

File: features/visitor_can_see_available_cars.feature [hunk 8]
```diff
@@ -0,0 +1,11 @@
+Feature: Visitor can see all cars available for rent
+
```
Comment (oliverochman) [inline features/visitor_can_see_available_cars.feature]: remove empty line

File: features/visitor_can_see_available_cars.feature [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    As a visitor
+    In order to rent a car
+    I want to see all the cars available for rent when I visit the site
+
+Scenario: View list of cars on landing page
```
Comment (oliverochman) [inline features/visitor_can_see_available_cars.feature]: Indent the scenario

File: app/views/car_card/index.html.haml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+%link{href: "https://cdn.jsdelivr.net/npm/tailwindcss/dist/tailwind.min.css", rel: "stylesheet"}/
+=render partial:"partials/nav"
+.max-w-md.w-full.lg:flex
+  .h-48.lg:h-auto.lg:w-48.flex-none.bg-cover.rounded-t.lg:rounded-t-none.lg:rounded-l.text-center.overflow-hidden{style: "background-image: url('https://images.unsplash.com/photo-1532974297617-c0f05fe48bff?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=700&q=80')", title: "Red car"}
+  .border-r.border-b.border-l.border-grey-light.lg:border-l-0.lg:border-t.lg:border-grey-light.bg-white.rounded-b.lg:rounded-b-none.lg:rounded-r.p-4.flex.flex-col.justify-between.leading-normal
+    .mb-8
+      .text-black.font-bold.text-xl.mb-2 BRAND MODEL
```
Comment (oliverochman) [inline app/views/car_card/index.html.haml]: you are not actually showing a car, you have hardcoded the stuff that you are supposed to see

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/views/car_card/index.html.haml (hunk 10 above)
Comment (Jon1969Edwards) [reply app/views/car_card/index.html.haml]: We know.  It's just a template that will be changed when the 'Add/Edit Car' page is done.

File: Gemfile [hunk 11]
```diff
@@ -14,6 +14,7 @@ gem 'jbuilder', '~> 2.5'
 gem 'bootsnap', '>= 1.1.0', require: false
 gem 'devise'
 gem 'haml'
+gem 'faker'
```
Comment (oliverochman) [inline Gemfile]: you are not using this gem anywhere

File: app/views/car_card/index.html.haml (hunk 10 above)
Comment (oliverochman) [reply app/views/car_card/index.html.haml]: You can still show the real car here. Add a background step in the scenario where you create the cars.

File: app/controllers/car_card_controller.rb (hunk 6 above)
Comment (johannaanden) [reply app/controllers/car_card_controller.rb]: Done

File: app/controllers/car_card_controller.rb (hunk 7 above)
Comment (johannaanden) [reply app/controllers/car_card_controller.rb]: Done

File: features/visitor_can_see_available_cars.feature (hunk 8 above)
Comment (johannaanden) [reply features/visitor_can_see_available_cars.feature]: Done

File: features/visitor_can_see_available_cars.feature (hunk 9 above)
Comment (johannaanden) [reply features/visitor_can_see_available_cars.feature]: Done

File: app/views/car_card/index.html.haml (hunk 10 above)
Comment (johannaanden) [reply app/views/car_card/index.html.haml]: Done

File: Gemfile (hunk 11 above)
Comment (johannaanden) [reply Gemfile]: Removed

File: app/controllers/automobiles_controller.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+class AutomobilesController < ApplicationController
+  def index
+    @automobiles = Automobile.all
+  end
+
+  def show
```
Comment (oliverochman) [inline app/controllers/automobiles_controller.rb]: you can remove this action because it is out of scope for this feature and you dont use it

**Review by oliverochman — APPROVED** [review]

**Pull request merged**